src/test/java/io/cdap/plugin/salesforce/etl/BaseSalesforceETLTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.salesforce.etl;
+
+import com.google.common.collect.ImmutableMap;
+import com.sforce.soap.partner.Error;
+import com.sforce.soap.partner.PartnerConnection;
+import com.sforce.soap.partner.SaveResult;
+import com.sforce.soap.partner.sobject.SObject;
+import com.sforce.ws.ConnectionException;
+import io.cdap.cdap.etl.mock.test.HydratorTestBase;
+import io.cdap.plugin.common.Constants;
+import io.cdap.plugin.salesforce.SalesforceConnectionUtil;
+import io.cdap.plugin.salesforce.SalesforceConstants;
+import io.cdap.plugin.salesforce.authenticator.AuthenticatorCredentials;
+import io.cdap.plugin.salesforce.plugin.ErrorHandling;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.rules.TestName;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Methods to run ETL with Salesforce Bulk/Streaming plugin as source, and a mock plugin as a sink.
+ *
+ * By default all tests will be skipped, since Salesforce credentials are needed.
+ *
+ * Instructions to enable the tests:
+ * 1. Create/use existing Salesforce account
+ * 2. Create connected application within the account to get clientId and clientSecret
+ * 3. Run the tests using the command below:
+ *
+ * mvn clean test
+ * -Dsalesforce.test.clientId= -Dsalesforce.test.clientSecret= -Dsalesforce.test.username= -Dsalesforce.test.password=
+ *
+ */
+public abstract class BaseSalesforceETLTest extends HydratorTestBase {
+
+  protected static final String CLIENT_ID = System.getProperty("salesforce.test.clientId");
+  protected static final String CLIENT_SECRET = System.getProperty("salesforce.test.clientSecret");
+  protected static final String USERNAME = System.getProperty("salesforce.test.username");
+  protected static final String PASSWORD = System.getProperty("salesforce.test.password");
+
+  protected static final String LOGIN_URL = System.getProperty("salesforce.test.loginUrl",
+                                                             "https://login.salesforce.com/services/oauth2/token");
+
+  @Rule
+  public TestName name = new TestName();
```

Comment:
nit: let's call this testName.

I was confused for a little bit about what this was when I first saw it being used in the tests.

Reply:
done